compiler/rustc_const_eval/src/interpret/intrinsics.rs

```diff
@@ -70,6 +70,9 @@ pub(crate) fn eval_nullary_intrinsic<'tcx>(
             ty::Pat(_, pat) => match **pat {
                 ty::PatternKind::Range { .. } => ConstValue::from_target_usize(0u64, &tcx),
                 // Future pattern kinds may have more variants
+                // FIXME(pattern_types): make this report the number of distinct variants used in the
+                // or pattern in case the base type is an enum.
+                ty::PatternKind::Or(_) => ConstValue::from_target_usize(0_u64, &tcx),
```

Comment:
Shouldn't there be a similar FIXME somewhere in codegen?

Reply:
codegen invokes this code path